### PR TITLE
Update public facing archer references to ecs-preview

### PR DESCRIPTION
**Pull request by sonofachamp**
* Rename the built binary from "archer" to "ecs-preview"
* Update front-facing references from "archer" to "ecs-preview"

**NOTE:** This doesn't update all references of "archer" to "ecs-preview" as many of those references are in test code or comments that won't show up in the UX. I can update those as well if desired.

#### Outstanding questions/changes
* Update SSM path values
  * If we can come up with binary-name agnostic path values that would be ideal so that we don't have to change them again unless we want them to match the binary name?
* Should we rename the `archer` package to something else?

Addresses #400 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/archer/main.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
```
Comment (hencrice) [inline cmd/archer/main.go]: should we remove this file?

File: NOTICE [hunk 2]
```diff
@@ -1,2 +1,2 @@
-PRIVATE Amazon Ecs Archer
+PRIVATE Amazon ECS ecs-preview
```
Comment (efekarakus) [inline NOTICE]: PRIVATE can be removed here

File: internal/pkg/deploy/cloudformation/env_test.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// Package cloudformation provides functionality to deploy archer resources with AWS CloudFormation.
+// Package cloudformation provides functionality to deploy ECS resources with AWS CloudFormation.
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/env_test.go]: nit: maybe `"... to deploy CLI concepts with ..."`, just ECS feels like it's to deploy things regarding only Amazon ECS.

**Comment by sonofachamp** [comment]
I originally missed the top-level `buildspec.yml`. If folks can take a look at that before I merge that'd be great.

**Pull request merged**